code/modules/mob/living/basic/pets/penguin.dm

```diff
@@ -0,0 +1,192 @@
+
+/mob/living/basic/pet/penguin/death(gibbed)
+	. = ..()
+	remove_egg()
+
+/mob/living/basic/pet/penguin/proc/lay_penguin_egg(obj/item/penguin_egg)
+	if(prob(35))
+		penguin_egg.AddComponent(\
+			/datum/component/fertile_egg,\
+			embryo_type = /mob/living/basic/pet/penguin/baby,\
+			minimum_growth_rate = 1,\
+			maximum_growth_rate = 2,\
+			total_growth_required = 400,\
+			current_growth = 0,\
+			location_allowlist = typecacheof(list(/turf, /mob/living/basic/pet/penguin)),\
+		)
+
+/mob/living/basic/pet/penguin/proc/on_hatch_egg(obj/source)
+	SIGNAL_HANDLER
+
+	remove_egg()
+	UnregisterSignal(source, COMSIG_QDELETING)
+
+/mob/living/basic/pet/penguin/proc/remove_egg()
+	if(carried_egg)
+		carried_egg.forceMove(get_turf(src))
+	carried_egg = null
+	cut_overlays()
+
+/datum/ai_controller/basic_controller/penguin
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic,
+	)
+
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/find_nearest_thing_which_attacked_me_to_flee,
+		/datum/ai_planning_subtree/flee_target,
+		/datum/ai_planning_subtree/find_and_hunt_target/penguin_egg,
+		/datum/ai_planning_subtree/random_speech/penguin,
+	)
+
+/datum/ai_planning_subtree/find_and_hunt_target/penguin_egg
+	target_key = BB_LOW_PRIORITY_HUNTING_TARGET
+	hunting_behavior = /datum/ai_behavior/hunt_target/penguin_egg
+	hunt_targets = list(/obj/item/food/egg/penguin_egg)
+	hunt_range = 7
+
+/datum/ai_behavior/hunt_target/penguin_egg
+	hunt_cooldown = 15 SECONDS
+	always_reset_target = TRUE
+
+/datum/ai_behavior/hunt_target/penguin_egg/target_caught(mob/living/basic/hunter, obj/item/food/egg/target)
+	hunter.UnarmedAttack(target, TRUE)
+
+/mob/living/basic/pet/penguin/emperor
+	name = "Emperor penguin"
+	real_name = "penguin"
+	desc = "Emperor of all she surveys."
+	icon_state = "penguin"
+	icon_living = "penguin"
+	icon_dead = "penguin_dead"
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+/mob/living/basic/pet/penguin/emperor/shamebrero
+	name = "Shamebrero penguin"
+	icon_state = "penguin_shamebrero"
+	icon_living = "penguin_shamebrero"
+	gold_core_spawnable = NO_SPAWN
+	unique_pet = TRUE
+
+/mob/living/basic/pet/penguin/baby
+	name = "Penguin chick"
+	real_name = "penguin"
+	desc = "Can't fly and barely waddles, yet the prince of all chicks."
+	icon_state = "penguin_baby"
+	icon_living = "penguin_baby"
+	icon_dead = "penguin_baby_dead"
+	density = FALSE
+	pass_flags = PASSMOB
+	mob_size = MOB_SIZE_SMALL
+	butcher_results = list(/obj/item/organ/internal/ears/penguin = 1, /obj/item/food/meat/slab/penguin = 1)
+	ai_controller = /datum/ai_controller/basic_controller/penguin/baby
+	can_lay_eggs = FALSE
+	///he will grow or not
+	var/grow_up = TRUE
+
+
+/mob/living/basic/pet/penguin/baby/Initialize(mapload)
+	. = ..()
+	if(!grow_up)
+		return
+	var/grown_type
+	if(prob(95))
```

Comment:
could be a nice weighted list like

```dm
var/list/potential_types = list(
     /path/one,
     /path/two,
)
```

then do `pick_weight()` on the list or something. i know i haven't done that in my refactors but it would be a nice pattern to start working on.